com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,202 @@
+import com.chartboost.heliumsdk.ad.HeliumFullscreenAd
+import com.chartboost.heliumsdk.ad.HeliumRewardedAd
+import com.unity3d.player.UnityPlayer
+
+class HeliumUnityAdWrapper(private val ad: HeliumAd) {
+    /**
+     * This is the container for the banner.
+     * We need a relative layout to position the banner in one of the 7 possible positions.
+     * HeliumBannerAd is just a ViewGroup.
+     * */
+    private var bannerLayout: RelativeLayout? = null
+    private val activity: Activity? = UnityPlayer.currentActivity
+
+    fun load() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            ad.load()
+        }
+    }
+
+    fun load(screenLocation: Int) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            createBannerLayout(screenLocation)
+            load()
+        }
+    }
+
+    fun show() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (ad is HeliumFullscreenAd) {
+                ad.show()
+            }
+        }
+    }
+
+    fun setKeyword(keyword: String, value: String): Boolean {
+        return ad.keywords.set(keyword, value)
+    }
+
+    fun removeKeyword(keyword: String): String {
+        return ad.keywords.remove(keyword) ?: ""
+    }
+
+    fun setCustomData(customData: String) {
+        if (ad is HeliumRewardedAd) {
+            ad.customData = customData
+        } else {
+            Log.d(TAG,"custom data can only be set on a rewarded ad")
+        }
+    }
+
+    fun setBannerVisibility(isVisible: Boolean) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (ad is HeliumBannerAd && bannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+                bannerLayout?.visibility = visibility
+                ad.visibility = visibility
+            }
+        }
+    }
+
+    fun clearLoaded(): Boolean {
+        if (ad is HeliumFullscreenAd) {
+            return ad.clearLoaded()
+        }
+        if (ad is HeliumBannerAd) {
+            HeliumUnityBridge.runTaskOnUiThread { ad.clearAd() }
+            return true
+        }
+        return false
+    }
+
+    fun readyToShow(): Boolean {
+        return try {
+            when (ad) {
+                is HeliumFullscreenAd -> ad.readyToShow()
+                is HeliumBannerAd -> {
+                    Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad.destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (ad !is HeliumBannerAd) {
+            return
+        }
+
+        if (activity == null)
+            return
```

Comment:
Log a reason explaining why? Also let's be consistent with the use of `{ }` for `if`.

Reply:
Yes